---
title: "GEO DNAm Data"
subtitle: "Extract all 450k, EPIC, EPICv2 datasets"
date: "`r format(Sys.time(), '%d %B %Y')`"
author: Paul Yousefi
output:
  html_document:
    css: docs/style.css
    toc: true
    highlight: tango
---

```{r setoptions, eval = TRUE, echo = FALSE}
knitr::opts_chunk$set(
comment = ">",
error = FALSE,
tidy = FALSE,
echo = TRUE, 
warning = FALSE, 
message = FALSE,
cache=FALSE)
```

```{r call_source, echo = F}
path <- "."
sourcecode <- grep("report.r$", 
	list.files(path), value = T)
read_chunk(file.path(path, sourcecode))
```

# Purpose

To identify and access all available DNAm datasets provided by the Gene Expression Omnibus (GEO) https://www.ncbi.nlm.nih.gov/geo/.  

While addition eligibility criteria are discussed further below, datasets needed to have been run in humans on an omic-level DNA methylation microarray (450k, EPIC, EPICv2) to have even been considered. 

```{r globals, results = 'hide'}
```

```{r load.data}
```


# Methods

Used functions included in the `geograbi` package to query and extract all data hosted on GEO

Used `geograbi.retrieve.datasets()` to query GEO for experiment records meeting the following selection criteria:

* Human subjects
* DNAm array data available on an omic-level platform (450k, EPIC, EPICv2)
* N >100

### Human DNAm Array experiments with N>100

N = `r nrow(gses)` experiments meeting our inclusion criteria

* Of those, N = `r sum(!is.na(gses$filenames))` had a GEO series matrix file, which contains all array, sample and series data for a given experiment, downloadable by `geograbi`

* N = `r sum(gses$class != "error")` series matrix files could be contacted and potentially extracted using `geograbi.get.data()`

	* Occasionally, methylation information is stored in a supplementary file, but retrieval of these was not explored. 
	
	* Manual exception processing can be performed but has not yet been implemented.

